### PR TITLE
Allow machine and test as query parameters for jobs

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: lib/OpenQA/Schema/ResultSet/Jobs.pm [hunk 1]
```diff
@@ -282,10 +282,12 @@ sub complex_query {
     }
     else {
         my %js_settings = map { uc($_) => $args{$_} } qw(iso hdd_1);
-        my $subquery = $schema->resultset("JobSettings")->query_for_settings(\%js_settings);
-        push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
+        if (%js_settings) {
+            my $subquery = $schema->resultset("JobSettings")->query_for_settings(\%js_settings);
+            push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
+        }
```
Comment (okurz) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm]: does a test cover the "else" path?

**Comment by okurz** [comment]
./t/04-scheduler.t complains now (from travis) that the worker id does not exist anymore.

**Comment by coolo** [comment]
yeah, because these *scheduler.t tests are morons :)

**Comment by coolo** [comment]
I punished them hard now - see cc443d2

File: lib/OpenQA/Schema/ResultSet/Jobs.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm]: tons actually. I just moved the if here as I noticed some queries that go like ... where job_id in (select job_id from job_settings) - which sounded a bit harsh to the database to me .

**Comment by aaannz** [comment]
LGTM

File: lib/OpenQA/Schema/ResultSet/Jobs.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm]: but I can't prove as the file is not listed in the coverage report

**Pull request merged**